app/src/main/java/org/oppia/android/app/home/classroomlist/ClassroomSummaryViewModel.kt

```diff
@@ -0,0 +1,44 @@
+package org.oppia.android.app.home.classroomlist
+
+import org.oppia.android.app.classroom.ClassroomListViewModel
+import org.oppia.android.app.home.HomeItemViewModel
+import org.oppia.android.app.model.ClassroomSummary
+import org.oppia.android.app.model.EphemeralClassroomSummary
+import org.oppia.android.domain.translation.TranslationController
+import java.util.Objects
+
+/** The view model corresponding to individual classroom summaries in the classroom summary RecyclerView. */
+class ClassroomSummaryViewModel(
+  classroomListViewModel: ClassroomListViewModel,
+  ephemeralClassroomSummary: EphemeralClassroomSummary,
+  translationController: TranslationController,
+) : HomeItemViewModel() {
+
+  /** Click listener for [ClassroomSummary] items. */
+  private val classroomClickListener = classroomListViewModel as ClassroomSummaryClickListener
+
+  /** The [ClassroomSummary] retrieved from the [EphemeralClassroomSummary]. */
+  val classroomSummary: ClassroomSummary = ephemeralClassroomSummary.classroomSummary
+
+  /** Lazy-loaded title extracted using the [TranslationController]. */
+  val title: String by lazy {
+    translationController.extractString(
+      ephemeralClassroomSummary.classroomSummary.classroomTitle,
+      ephemeralClassroomSummary.writtenTranslationContext
+    )
+  }
+
+  /** Handles the click event for a [ClassroomSummary] by invoking the click listener. */
+  fun handleClassroomClick(classroomSummary: ClassroomSummary) {
```

Comment:
```suggestion
  fun handleClassroomClick() {
```
No need to pass in `classroomSummary` as it is available in this class.

Reply:
Done.